### PR TITLE
[ricos-editor] added floating plugin menu button to tiptap

### PR DESCRIPTION
**Pull request by tamrat-bay**

### CODE REVIEW

File: packages/ricos-editor/web/src/RicosEditorTiptap/FullRicosEditorTiptap.tsx [hunk 1]
```diff
@@ -18,10 +25,15 @@ class FullRicosEditorTiptap extends React.Component<RicosEditorProps> {
 
   content = Content.create<Node[]>([]);
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (tamrat-bay) [inline packages/ricos-editor/web/src/RicosEditorTiptap/FullRicosEditorTiptap.tsx:28]: remove

File: packages/ricos-editor/web/src/utils/pluginMenuUtils.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const posToDOMRect = (view: any, from: number, to: number): DOMRect => {
```
Comment (tamrat-bay) [inline packages/ricos-editor/web/src/utils/pluginMenuUtils.ts:2]: move out 
  function minMax(value = 0, min = 0, max = 0): number {
    return Math.min(Math.max(value, min), max);
  }

File: packages/tiptap-editor/web/src/components/RicosTiptapEditor/RicosTiptapEditor.tsx [hunk 3]
```diff
@@ -14,6 +14,7 @@ import { patchExtensions } from '../../patch-extensions';
 import '../../statics/styles/tiptap-editor-styles.scss';
```
Comment (danarish4) [inline packages/tiptap-editor/web/src/components/RicosTiptapEditor/RicosTiptapEditor.tsx:14]: revert file

File: packages/ui-components/web/src/Icons/CirclePlusIconSmall.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable max-len */
```
Comment (danarish4) [inline packages/ui-components/web/src/Icons/CirclePlusIconSmall.tsx:1]: remove for now

File: packages/ricos-editor/web/src/RicosEditorTiptap/FullRicosEditorTiptap.tsx [hunk 5]
```diff
@@ -5,8 +5,15 @@ import RicosEditorTiptap from './RicosEditorTiptap';
...
 import { RicosContextProvider } from '../RicosContext/RicosContext';
 import type { Node } from 'prosemirror-model';
+import PluginMenuButton from './PluginMenuButton';
+import { posToDOMRect } from '../utils/pluginMenuUtils';
 
-class FullRicosEditorTiptap extends React.Component<RicosEditorProps> {
+interface State {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  pluginMenuPosition: Record<string, any> | undefined;
+}
```
Comment (danarish4) [inline packages/ricos-editor/web/src/RicosEditorTiptap/FullRicosEditorTiptap.tsx:14]: move this logic outside from rocis full editor to add plugin component

File: packages/ricos-editor/web/src/utils/pluginMenuUtils.ts (hunk 2 above)
Comment (danarish4) [inline packages/ricos-editor/web/src/utils/pluginMenuUtils.ts:2]: please align file name with func name

File: packages/ricos-editor/web/src/RicosEditorTiptap/FullRicosEditorTiptap.tsx [hunk 6]
```diff
@@ -102,6 +103,7 @@ class FullRicosEditorTiptap extends React.Component<RicosEditorProps> {
                 content={this.content}
                 editorCommands={this.editor.commandManager}
               />
+              <PluginMenuButton editor={this.editor} />
```
Comment (danarish4) [inline packages/ricos-editor/web/src/RicosEditorTiptap/FullRicosEditorTiptap.tsx:106]:
```suggestion
              <FloatingAddPluginMenu editor={this.editor} />
```

File: packages/ricos-editor/web/statics/styles/plugin-menu.scss [hunk 7]
```diff
@@ -0,0 +1,8 @@
+.pluginMenu_wrapper {
```
Comment (danarish4) [inline packages/ricos-editor/web/statics/styles/plugin-menu.scss:1]:
```suggestion
.floatingAddPluginMenu_wrapper {
```

File: packages/ricos-editor/web/statics/styles/plugin-menu.scss [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+  position: relative;
+}
+
+.pluginMenu_button {
```
Comment (danarish4) [inline packages/ricos-editor/web/statics/styles/plugin-menu.scss:5]:
```suggestion
.floatingAddPluginMenu_button {
```

**Comment by oded-soffrin** [comment]
🤩

**Pull request merged**